src/lsm/lsm_meta.c

```diff
@@ -349,6 +349,13 @@ __lsm_meta_read_v1(
 	 * Ignore any other values: the metadata entry might have been
 	 * created by a future release, with unknown options.
 	 */
+	/*
+	 * If the default merge_min was not overridden, calculate it now.  We
+	 * do this here so that trees created before merge_min was added get a
```

Comment:
I think this change is correct, but I don't think the comment is accurate. The default merge_min is 0 - because we want the default to be half of merge_max (i.e: needing to do this calculation is expected). We could go in a couple of different directions here:
* Update the comment and leave the behavior as is.
* Change the merge_min default to be 7 (half the default merge_max).
* Ensure that a merge_min is added to the configuration in ``__wt_lsm_meta_write`` so it will always be present.

I'm inclined to go with the first option. The second option avoids this same error happening in the future, but leads to an API change.